### DISCUSSION TITLE
Incorrect type. Expected one of array, null.yaml-schema: Ansible Tasks Schema - .moon/tasks/*.yml

### DISCUSSION THREAD

**Discussion by nhattran998** [moonrepo/moon#1441] [Q&A] [answered]
Hi Moon Team, thank you for the amazing tools - I fell in love with this + proto, but we had a problem with the .yml file.
the tasks .yml files are always red, with a warning like the title. So what we can do with that? I tried VsCode or Zed, both claim the same thing.

<img width="820" alt="image" src="https://github.com/moonrepo/moon/assets/24296018/df344d8c-2602-4a94-948f-a75e2a43e27b">

**Comment by milesj** [comment]
@harrytran998 I don't have this issue, so something else seems to be going on. Maybe conflicting extension? I'm not entirely sure what "ansible tasks schema" is?

**Reply by nhattran998** [reply]
<img width="691" alt="image" src="https://github.com/moonrepo/moon/assets/24296018/c4759136-37f6-42f9-9a4c-a9b231066f46">

Yeah, in VsCode --> After I disable the extensions `https://github.com/redhat-developer/vscode-yaml.git` --> not see the error anymore. But It will be lost every suggestion of the yml file although I put the `$schema: 'https://moonrepo.dev/schemas/tasks.json'` at the top of the file.

In Zed Editor, I am not installing any extensions--> This is the default behavior.
@milesj Can you need to provide more information?

**Reply by milesj** [reply]
@harrytran998 Do you use ansible lint? https://github.com/ansible/ansible-lint/discussions/1399

They have a glob for `"tasks/*.yml"` which looks like it conflicts with moon. That's why you see "Ansible tasks schema" because it's using that json schema, and not actually moons.

**Reply by milesj** [reply, marked as answer]
At the bottom of vscode, you should be able to choose which schema I believe.
<img width="721" alt="Screenshot 2024-04-21 at 12 06 55 PM" src="https://github.com/moonrepo/moon/assets/143744/32006d2b-d791-43f6-b394-c6e3873a7b86">

**Comment by nhattran998** [comment]
@milesj Quite strange, I did not use Ansible + I can not choose Schema Moon like you 🤔. 
<img width="618" alt="image" src="https://github.com/moonrepo/moon/assets/24296018/f6af38d3-1b37-4760-8e5d-00404e9e6a32">


Here is my VsCode info:

```
Version: 1.88.1 (Universal)
Commit: e170252f762678dec6ca2cc69aba1570769a5d39
Date: 2024-04-10T17:42:52.765Z
Electron: 28.2.8
ElectronBuildId: 27744544
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Darwin x64 22.6.0
```

**Reply by milesj** [reply]
I see "Ansible Tasks schema" in that list, so it's probably falling back to that.

Dumb question, but do you have the moonrepo vscode extension installed?

**Reply by nhattran998** [reply]
Oh, after I installed it, seem VsCode does not give me the choice to unselect Ansible Task, but I can uncheck the moon repo
<img width="670" alt="image" src="https://github.com/moonrepo/moon/assets/24296018/01d46910-f494-4407-8f85-14bc3d7987fd">

**Reply by nhattran998** [reply]
Hmm, seems after I restart my computer, VsCode detects this yml file correctly. So thank you for your time 🙏.

**Reply by milesj** [reply]
Glad you figured this out!
